### ISSUE TITLE
add training loop to Trainer

### ISSUE THREAD

**Issue by KappaDistributive** [riven314/Controllable-Image-Captioning-App#24]
Add a training loop (with loss function, optimizer, ...) to the Trainer.
